src/modules/shlvl.rs

```diff
@@ -0,0 +1,54 @@
+use std::env;
+
+use super::{Context, Module, SegmentConfig};
+
+use crate::config::RootModuleConfig;
+use crate::configs::shlvl::ShLvlConfig;
+
+const SHLVL_ENV_VAR: &str = "SHLVL";
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let shlvl = get_shlvl_value();
+
+    match shlvl {
+        Some(shlvl_num) => {
+            let mut module = context.new_module("shlvl");
+            let config: ShLvlConfig = ShLvlConfig::try_load(module.config);
+
+            if config.disabled {
+                return None;
+            }
+
+            if config.threshold <= shlvl_num {
+                module.set_style(config.style);
+
+                module.get_prefix().set_value(config.prefix);
+                module.get_suffix().set_value(config.suffix);
+
+                if let Some(symbol) = config.symbol {
+                    module.create_segment("symbol", &symbol);
+                }
+
+                module.create_segment("value", &SegmentConfig::new(&shlvl_num.to_string()));
+
+                Some(module)
+            } else {
+                None
+            }
+        }
+        None => None,
+    }
+}
+
+fn get_shlvl_value() -> Option<i64> {
```

Comment:
Ok, a few notes. If you are going to convert from OsString to String, you might as well use `env::var` rather than `env::var_os`, which does that conversion for you.

I believe you should be able to write the whole function as:
`env::var(SHLVL_ENV_VAR).ok()?.parse::<i64>().ok()`

(But I didn't try to compile this, I might have missed something subtle.)